pandas/_libs/reduction.pyx

```diff
@@ -518,6 +519,7 @@ def apply_frame_axis0(object frame, object f, object names,
         except:
             raise InvalidApply('Let this error raise above us')
 
+    piece = result
```

Comment:
The ``try...except`` above is based on the assumption that ``result = f(chunk)`` can fail. I have no idea of when/why it does, but if it does, ``result`` is not defined and this line fails.

Reply:
I assumed if exception is raised, program terminates and wouldn't reach this line. To be safe I'll move it inside the try clause, or maybe should rename result to piece

Reply:
Sorry, you are absolutely right. What I should have written is: this will fail if ``n=0``. Then I have no idea if it can happen, but the code that follows doesn't fail.